types/src/transaction/mod.rs

```diff
@@ -569,19 +569,26 @@ pub enum TransactionStatus {
 
     /// Keep the transaction output
     Keep(VMStatus),
+
+    /// Retry the transaction because it is after a ValidatorSetChange txn
+    Retry,
 }
 
 impl TransactionStatus {
-    pub fn vm_status(&self) -> &VMStatus {
+    pub fn vm_status(&self) -> VMStatus {
         match self {
-            TransactionStatus::Discard(vm_status) | TransactionStatus::Keep(vm_status) => vm_status,
+            TransactionStatus::Discard(vm_status) | TransactionStatus::Keep(vm_status) => {
+                vm_status.clone()
+            }
+            TransactionStatus::Retry => VMStatus::new(StatusCode::UNKNOWN_VALIDATION_STATUS),
```

Comment:
should this be unreachable?

Reply:
I think this line is reachable if `vm_status` is a method of `TransactionStatus`